fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/mixin/gametest/StructureTestUtilMixin.java

```diff
@@ -38,26 +44,51 @@
 public abstract class StructureTestUtilMixin {
 	private static final String GAMETEST_STRUCTURE_PATH = "gametest/structures/";
 
-	// Replace the default test structure loading with something that works a bit better for mods.
+	// Use "gametest/structures/" as default test structure directory for ".snbt" files. Fall back to vanilla if not found.
 	@Inject(at = @At("HEAD"), method = "createStructure(Ljava/lang/String;Lnet/minecraft/server/world/ServerWorld;)Lnet/minecraft/structure/Structure;", cancellable = true)
 	private static void createStructure(String id, ServerWorld world, CallbackInfoReturnable<Structure> cir) {
 		Identifier baseId = new Identifier(id);
 		Identifier structureId = new Identifier(baseId.getNamespace(), GAMETEST_STRUCTURE_PATH + baseId.getPath() + ".snbt");
+		String snbt = null;
 
 		try {
 			Resource resource = world.getServer().getResourceManager().getResource(structureId);
-			String snbt;
 
 			try (InputStream inputStream = resource.getInputStream()) {
 				snbt = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
 			}
+		} catch (IOException ignore) {
+			// Ignore
```

Comment:
Perhaps explain why you are ignoring it, or just log an error so you avoid crashing?

Reply:
An error message which combines an error for all three tries to find the structures is handled here:
https://github.com/FabricMC/fabric/pull/2163/files#diff-68f999bc4b47c492f226b5a47e7b54b644846784de57e477fff655c6ebefb2cdR89

If we would throw the exception there, we would not even try to find the structure in the `StructureManager`.  This is currently the behavior in fabric, which only tries to read from `/gametest/structure/*.snbt` and throws the IOException and exit the process.